Parse documents into a structure good for the rest of the application

gwitmond opened the pull request:
It manages a queue in the DB.

The API offers an Enqueue method to be used in the Document uploaders interface.
And an RunQueue method to be used at the server that does the parsing.

gwitmond committed:
Adding dependencies Leptonica and Tesseract.

gwitmond committed:
Added letterbox.go. A queue managent and parser for documents.

GeertJohan commented on gopath/src/github.com/GeertJohan/go.leptonica/leptonica.go:
```diff
@@ -0,0 +1,51 @@
+#include "leptonica/allheaders.h"
+#include <stdlib.h>
+
+l_uint8* uglycast(void* value) { return (l_uint8*)value; } 
```
Maybe rename uglycast to something more descriptive?

gwitmond committed:
Merge branch 'master' of github.com:nulpunt/nulpunt

gwitmond committed:
Intitial proposal for the Server API.
It describes the way the application must react to input.
It is the base of the application design between the front end and the back end. Any deviations between the document and reality should be considered to be bugs. Solve them!

gwitmond committed:
Better formatting for readability.
